config/configgrpc/configgrpc.go

```diff
@@ -259,8 +259,7 @@ func (gcs *GRPCClientSettings) toDialOptions(host component.Host, settings compo
 	}
 
 	// Enable OpenTelemetry observability plugin.
-	opts = append(opts, grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor(otelOpts...)))
-	opts = append(opts, grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor(otelOpts...)))
+	opts = append(opts, grpc.WithStatsHandler(otelgrpc.NewClientHandler(otelOpts...)))
```

Comment:
Finally they did it.

Reply:
Yeah although based on the results, something doesn't look right: https://github.com/open-telemetry/opentelemetry-collector/actions/runs/6853173539/job/18633503149?pr=8855

Reply:
@bogdandrutu ah nevermind, i see the test is just checking the len of the opts 🤣 will submit a PR to fix it

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/pull/8864